File: problem2.cpp
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

class Solution {
public:
ListNode* removeNthFromEnd(ListNode* head, int n) {
        
        ListNode *slow = head, *fast = head;

        for(int i=0;i<n;i++) 
            fast = fast->next;
        
        if (!fast) 
            return head->next;

        while (fast->next) 
        {
            slow = slow->next;
            fast = fast->next;
        }

        ListNode* toDelete = slow->next;

        slow->next = slow->next->next;

        delete toDelete;

        return head;
    }
};

File: Problem2.js
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no


// Your code here along with comments explaining your approach
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
 var removeNthFromEnd = function(head, n) {
    let fast = head, slow = head;
    let prev = 0;
    for(let i = 0; i<n; i++){
        fast = fast.next;
    }
    
    while(fast){
        prev = slow;
        slow = slow.next;
        fast = fast.next;
    }
    
    if(!prev) return slow.next;
    prev.next = slow.next;
    return head;
    
};